### ISSUE TITLE
Don't allow any mappers to set squares as "Unavailable" except validators or project managers

### ISSUE THREAD

**Issue by aawiseman** [hotosm/tasking-manager#5389]
Often mappers will mark many many squares as "unavailable", maybe when they are slightly blurry but still find to map. This means many squares don't get mapped and people have to ask the tech team or project managers to change the square status back to "ready to map". This is time consuming and not efficient.

Let's take away the possibility to mark squares as "unavailable" to anyone except validators and project managers. 

People can already comment in the squares, so if they think the imagery is unusable you can comment there.

Ralph from the Data Quality Working Group reported that there were multiple projects with 50-80 squares marked unavailable where he had to go to the TM team to get the squares changed.

**Comment by HelNershingThapa** [comment]
Hi @aawiseman, I looked into the problem and discovered that only users with advanced mapping capabilities are allowed to mark a task as having bad imagery. But there is an error in the implementation: the radio button to offer the status as bad imagery is displayed to all users--this is unexpected. In contrast, it should only be shown to those with advanced mapping capabilities. Do you believe this issue would be remedied if this implementation was corrected? Or would you want to limit it to validators and project managers, as you've mentioned?

**Comment by ghimiresushma99** [comment]
Limiting the marking bad imagery feature to the advanced mappers or project managers/validators might not remedy the problem we are looking into. Since, mappers are the ones who inspect the tasks and look out for the things to map in them first, neither the validator nor the project manager. The validators come into action only after the mapper updates the status of the tasks. And only allowing advanced mappers to update the bad imagery status of the task will break the general loop of our mapping - validation for the rest of the mappers. What are intermediate mappers supposed to do when they come across tasks with cloud-covered imagery?

In addition, I personally find the issue of beginners marking "Bad Imagery" similar to marking tasks as "Completed" when there is still a lot to map in it. When beginners mark incomplete tiles as done, validators invalidate the task status and reopen it for mapping. The mapper receives the email and is often responsible for mapping the task again with the help of instructions/comments left by the validators. 

A similar approach can be adopted. When the wrongly marked bad imagery tasks are reverted, the mapper shall receive the notification and will work on it. I believe the new mappers don't get notified that their judgement about the image needs to be reviewed, and that's why keep on marking tasks with Blurry Images as "Bad Imagery". But who shall own the authority to revert "Bad-Imagery" labelled task and open it for mapping might be a discussion instead? Project Managers or Validators or both?

**Comment by martien-176** [comment]
I think the biggest issue here is that marked tasks are removed from the general mapping loop. That is also what @ghimiresushma99 points at, i believe.

So why not keep it simple and just remove the block functionality?

All mappers can mark tasks as having bad imagery. And these tasks are visible as such on the project map. But all mappers can also keep mapping the task if they feel comfortable or are able to. The marking is just a warning.

**Comment by ghimiresushma99** [comment]
Agreed with @martien-176 . Block functionality on "Bad-Imagery" tasks is creating an extra hurdle for the validators. Removing block functionality will let the marking be just a warning for validators. Validators can now easily revert the wrongly labelled task status.

**Comment by Ichchhie** [comment]
I totally agree with @ghimiresushma99 and @martien-176. After reading all this, my opinion is,  once the task is marked as bad imagery and goes to "unavailable" status, the particular task is only accessible to validators and not mappers so that validators can at least go into the task and check if that's actually an issue with imagery. 

- if there is an imagery issue, can revert the "Bad-Imagery" labeled task and reopen it for mapping 
- else, can work on it and submit the task.

**Comment by martien-176** [comment]
Hi @Ichchhie , blocking access to only validators can imo be sometimes harmfull. This only works if validators are available and willing to keep an eye on a particular project. That is not always the case and can lead to a piling up of unavailable tasks. While maybe there are still mappers who want to work on them.

What do you think of the idea to not make the marked tasks unavailable, but keep them open for all mappers? Consider the marking just as a "warning"? That way mappers can still work on them if they feel comfortable with it, eg by using alternative imagery. Or by playing around with their editor's imagery settings.

My impression is that there are not that many tasks anymore that are completely unusable due to clouding or blurry imagery. There is often alternative imagery. And, according to Ralph Aytoun, imagery companies now filter out the really unusable tiles beforehand and replace them with better (be it older) ones.

**Comment by martien-176** [comment]
As an additional benefit of having these "warnings" in place: that way project management / validators can see where projectimagery was problematic. 

And thus they know were to put some extra effort to check data quality.

**Comment by ghimiresushma99** [comment]
I agree with @martien-176  as well. Access only to the validators might result in the piling up of unavailable tasks. Instead of restricting access to unavailable labelled tasks, making them available for validation to all and treating the marking as a warning provides a better remedy.

**Comment by petya-kangalova** [comment]
@aawiseman @zlavergne @Ichchhie  tried to summarise the discussion in the meet up. Please add/ edit:

@aawiseman  maybe something to run via the data quality working group:

 1. Instead of having the "Yes", "No" and "Bad Imagery" options, we keep only two options " Yes" and "No". With the "No" having sub-options for the user to explain why they have selected "No".
 2.  Improve the option/ reasons for selecting "No". Some suggestions:

- Imagery Too Blurry
- This is too hard for me  to map [ this could be helpful to review the mapper level set]

Other suggestions are welcome! Also if I have missed any in the discussion.

**Comment by aawiseman** [comment]
@petya-kangalova that's a good idea, let me ask! Or feel free to join us on Slack too!

**Comment by russdeffner** [comment]
I like the idea but wonder if we try to limit the options but suggest they put a comment why the answer is "No". My concern is that in order to properly understand what the issue might be we made need a handful of imagery issue selections (examples: blurry, not loading, offset (and don't know how to correct), coloration and/or brightness, better imagery available vs. what is the default, expired API keys, imagery has recently changed, etc.). And then another list for why they stopped mapping (out of time, too difficult, didn't know how to interpret the imagery, etc.).

So, maybe we would have just 2-3 options (issue with the project, issue with the imagery, issue with editing) or alternatively, we come up with a 'short-list' of say 5 or fewer options with one being "Other" and have a bit more of a 'required' text box/comment.

**Comment by aawiseman** [comment]
@russdeffner I think that's a good idea to have specific things to choose

**Comment by RAytoun** [comment]
To put things in better perspective of the projects I have checked (this is not all projects) **78 Projects had a total of 1759 tasks** marked as **Unavailable**. In some instances this is a significant amount of the project as in **#13407 with 92 tasks = 17% of the project.** Many are in the range of around 6% of the project. Some of the projects I have already reverted the unavailable tasks such as #13280 with 233 tasks unavailable.

Most of these tasks are marked as such by beginner mappers who have not continued mapping and therefore will not benefit from feedback and go back to complete the task. Validators understandably do not have the permission or ability to revert all of the tasks in one go and this means the project author or myself will have to do this. Considering the volumes involved it takes up a lot valuable time checking if the imagery is bad before reverting all of them for each individual project.

Experience also tells me that making the options more complex is only going to make it even more difficult. We are dealing with new starters that do not yet understand all the complexities of imagery or the Tasking Manager, and as we have seen where the presets make mappers think they need to fill in the requested tags and they give things such as _name=house_ or _name=road_ or adding _layer=1_ to overlapping buildings.

I would like to see the ability to mark tasks as unavailable removed from the Mapping stage so that validators can open them and check the few features that have been added or do what beginners are still not familiar with - check other available imagery to see if there is other options. Then the validator can mark it unavailable so that project managers can see how much of the area was not mapped due to poor imagery.

If it is at all possible I would like to see the Bad Imagery tab removed from the task at mapping stage and only added after a task has been marked as completely mapped ready for validation.

Thank you.

**Comment by ramyaragupathy** [comment]
Related to this, we are starting with a [feature to let project managers/admins revert unavailable tasks marked by a specific user](https://github.com/hotosm/tasking-manager/issues/5943). This came up during a slack conversation with Dinar and Sabine.
